x/ccv/provider/types/keys.go

```diff
@@ -281,3 +245,83 @@ func HashString(x string) []byte {
 	hash := sha256.Sum256([]byte(x))
 	return hash[:]
 }
+
+// tsAndChainIdKey returns the key with the following format:
+// bytePrefix | len(timestamp) | timestamp | chainID
+func tsAndChainIdKey(prefix byte, timestamp time.Time, chainID string) []byte {
```

Comment:
A small nitpick about this and other functions we use to create keys.

Sometimes using `copy` with array/slice contatenation can be difficult to read. To me it seems like there's a lot more going on with the code than there actually is.

In this function, all values are simply appended to a byte slice.

Maybe we could make the code a bit more readable and reduce the number of comments by doing something like this:

```golang
func tsAndChainIdKey(prefix byte, timestamp time.Time, chainID string) []byte {
	timeBytes := sdk.FormatTimeBytes(timestamp)
	lenTimeBytes := sdk.Uint64ToBigEndian(uint64(len(timeBytes)))
        lenKey := len([]byte{prefix})+8+len(timeBytes)+len(chainID)
	
        // allocates array but does not populate with zero vals
	bz := make([]byte, 0, lenKey)
	bz = append(bz, []byte{prefix}...)
	bz = append(bz, lenTimeBytes...)
	bz = append(bz, timeBytes...)
	bz = append(bz, chainID...)
	return bz
}
```

With this, it is immediately apparent what is going on and there's no question about the order of the copied values.

Working example:
https://go.dev/play/p/m8ZkY8t6LCE

Reply:
Fine by me. I just used the existing model. If nobody objects, I can do the change.